inject/android.js

```diff
@@ -16,10 +25,48 @@ function initCode () {
         platform: processedConfig.platform
     })
 
+    const messageSecret = processedConfig.messageSecret
+    // Receives messages from the platform
+    const messageCallback = processedConfig.messageCallback
+    // Sends messages to the platform
+    const messageInterface = processedConfig.messageInterface
+
+    const wrappedUpdate = ((providedSecret, ...args) => {
+        if (providedSecret === messageSecret) {
+            update(...args)
+        }
+        // eslint-disable-next-line no-extra-bind
+    }).bind(this)
```

Comment:
@kzar I've suppressed this: `ESLint: The function binding is unnecessary`.
Let me know if this is the correct approach.

Reply:
Just remove it, arrow functions can't be bound.

Reply:
[See the note in the Spec](https://262.ecma-international.org/6.0/#sec-arrow-function-definitions-runtime-semantics-evaluation) _"An ArrowFunction does not define local bindings for arguments, super, this, or new.target. Any reference to arguments, super, this, or new.target within an ArrowFunction must resolve to a binding in a lexically enclosing environment."_

Even if you bind it's not possible, this is always the enclosed scope and the toString is never different:
```js
const other = {
  toString() {return 'meep'}
}
function beep() {
  const x = (() => {
    console.log('x', this)
  }).bind(other)
  console.log(1, x, x.toString(), x())
  const y = (() => {
    console.log('y', this)
  }).bind(other)
  console.log(2, y, y.toString(), y())
}
beep.prototype.toString = function () {
  return 'xxx'
}
beep()
```

> The bind here helps avoid wrappedUpdate.toString() revealing the messaging secret, which is worth doing.

Can you explain what you mean?

Reply:
> Can you explain what you mean?

If the website does `window[messageCallback].toString()` (after figuring out what `messageCallback` is), we don't want the value of `messageSecret` to be revealed. Using bind like I'm recommending avoids that:

```javascript
const example = ((providedSecret, ...args) => {
        if (providedSecret === messageSecret) {
            update(...args)
        }
        // eslint-disable-next-line no-extra-bind
    })
example.toString()

const wrappedExample = example.bind(this)
wrappedExample.toString()
```

Reply:
Ah! Cool thanks for explaining.

I think because bind is invalid you're essentially doing `.bind(undefined)` the lint still fails using that though 🥲